lib/src/web_socket/common/web_socket.dart

```diff
@@ -138,7 +140,9 @@ abstract class CommonWebSocket extends Stream implements WebSocket {
   /// result in a [StateError].
   @override
   Future<Null> addStream(Stream stream) async {
-    return _outgoing.addStream(stream);
+    _inProgressAddStream = _outgoing.addStream(stream);
+    await _inProgressAddStream;
+    _inProgressAddStream = null;
```

Comment:
This looks like an opportunistic bug fix where we we're awaiting.. is that accurate?